tests/e2e/redis/parallel-ingest-redis-serving.py

```diff
@@ -0,0 +1,185 @@
+PROJECT_NAME = "basic_" + uuid.uuid4().hex.upper()[0:6]
+
+
+@pytest.fixture(scope="module")
+def client(pytestconfig):
+    core_url = pytestconfig.getoption("core_url")
+    serving_url = pytestconfig.getoption("serving_url")
+
+    client = Client(core_url=core_url, serving_url=serving_url,)
+
+    client.set_project(PROJECT_NAME)
+
+    return client
+
+
+@pytest.mark.incremental
+class TestBasicIngestionRetrieval:
+    def setup_class(cls):
+        prefix = "basic_ingestion"
+        suffix = str(int(datetime.now().timestamp()))
+        cls.customer_ft_name = f"{prefix}_customer_{suffix}"
+        cls.driver_ft_name = f"{prefix}_driver_{suffix}"
+
+        cls.customer_entity = Entity(
+            name="customer_id",
+            description="Customer entity for rides",
+            value_type=ValueType.STRING,
+            labels={"team": "customer_service", "common_key": "common_val"},
+        )
+
+        cls.driver_entity = Entity(
+            name="driver_id",
+            description="Driver entity for car rides",
+            value_type=ValueType.STRING,
+            labels={"team": "matchmaking", "common_key": "common_val"},
+        )
+
+        cls.basic_ft_spec = FeatureTable.from_yaml(
+            f"{DIR_PATH}/specifications/dev_ft.yaml"
+        )
+
+    def test_discovery(self, client):
+
+        # ApplyEntity
+        client.apply_entity(self.customer_entity)
+        client.apply_entity(self.driver_entity)
+
+        # GetEntity Check
+        assert client.get_entity(name="customer_id") == self.customer_entity
+        assert client.get_entity(name="driver_id") == self.driver_entity
+
+        # ListEntities Check
+        common_filtering_labels = {"common_key": "common_val"}
+        matchmaking_filtering_labels = {"team": "matchmaking"}
+
+        actual_common_entities = client.list_entities(labels=common_filtering_labels)
+        actual_matchmaking_entities = client.list_entities(
+            labels=matchmaking_filtering_labels
+        )
+        assert len(actual_common_entities) == 2
+        assert len(actual_matchmaking_entities) == 1
+
+        # ApplyFeatureTable
+        client.apply_feature_table(self.basic_ft_spec, PROJECT_NAME)
+
+        # GetFeatureTable Check
+        actual_get_feature_table = client.get_feature_table(name="dev_featuretable")
+        assert actual_get_feature_table == self.basic_ft_spec
+
+        # ListFeatureTables Check
+        actual_list_feature_table = [
+            ft for ft in client.list_feature_tables() if ft.name == "dev_featuretable"
+        ][0]
+        assert actual_list_feature_table == self.basic_ft_spec
+
+    def test_basic_retrieval(self, client):
+        # TODO: Add ingest and retrieval check
+        pass
+
+
+@pytest.mark.incremental
+class TestAllTypesIngestionRetrieval:
+    def setup_class(cls):
+        prefix = "alltypes_ingestion"
+        suffix = str(int(datetime.now().timestamp()))
+        batch_source = DataSource(
```

Comment:
Is this setup the way you think users of Feast will end up using the API?

Reply:
This way, we can isolate the objects we need for this test suite in the `setup_class` method. Using classes also allow us to more easily plan which tests to run in sequence and which tests to run in parallel.

Reply:
The API should match commands that users are running. Is there any reason we cant just use a single big function if we want to run it in sequence?

Reply:
Also, `SourceType(1).name` doesnt look like something we would expose to users.

Reply:
Removed `SourceType(1).name`.

Refactored pytests to use fixtures instead of class-based approach.